posts/2018-02-09-serverless-and-graphql-in-love.md

```diff
@@ -0,0 +1,611 @@
+
+In this post, I’ll cover how to create Serverless GraphQL endpoints using DynamoDB, RDS and Twitter REST API.
+
+Ready? Awesome. Let's go. 
+
+## What is GraphQL?
+
+I’m going to start this off by stating a fact: The way we currently build APIs, as a collection of micro-services that are all split up and maintained separately, isn’t optimal. If you're a fellow back-end or front-end engineer, you're probably familiar with this struggle.
+
+Luckily for us, the tech horizon is ever-expanding. We have options. And we should use them.
+
+[GraphQL](https://dev-blog.apollodata.com/2017-the-year-in-graphql-124a050d04c6) lets you shrink your multitude of APIs down into a single HTTP endpoint, which you can use to fetch data from multiple data sources. In short, it provides a simple way of building mobile/web applications by providing a clean layer of abstraction between servers and clients.
+
+It lets you:
+1. Reduce network costs and get better query efficiency.
+2. Know exactly what your response will look like and ensure you're never sending more or less than the client needs.
+3. Describe your API with types that map your schema to existing backend.
+
+Thousands of companies are now using GraphQL in production with the help of open source frameworks built by Facebook, Apollo, and Graphcool. In fact, that [Starbucks announcement](https://twitter.com/davidbrunelle/status/960946257643454464) last week made my morning coffee taste even better. 😉
+
+![](https://user-images.githubusercontent.com/1587005/36067514-503a05b4-0e73-11e8-9b40-946c5398f4b5.png "Danielle's slide from Serverless and GraphQL meetup at Glassdoor, Jan 29, 2018")
+
+Now, very reasonably, you are probably thinking, “Yeah, okay, Facebook is one thing; they have a giant engineering team. But for me, having only one API endpoint is too risky. What if it goes down? How do I handle that much load? What about security?”
+
+You are absolutely correct: with one HTTP endpoint, you need be be entirely sure that endpoint never goes down and that it scales on demand.
+
+That’s where serverless comes in.
+
+## What is Serverless?
+
+Serverless has gained popularity over last few years, primarily because it gives developers flexibility.
+
+With Serverless comes the following:
+1. No server management (no need to manage any form of machine)
+2. Pay-per-execution (never pay for idle)
+3. Authentication and authorization at scale
+4. Function as a unit of application logic
+
+## What makes Serverless and GraphQL such a great fit?
+
+When moving to GraphQL, you suddenly rely on one HTTP endpoint to connect your clients to your backend services. Once you do decide to do that, you want this one HTTP endpoint to be: reliable, fast, auto-scaling, and have a small attack vector regarding security.
+
+*All* these properties are fulfilled by a single AWS Lambda function in combination with API Gateway. It’s just a great fit! 
```

Comment:
`with API Gateway` --> `with an API Gateway` (since you could use other API Gateways from e.g. Google as well)?